src/components/Cart/cartContext.js

```diff
@@ -0,0 +1,26 @@
+import { createContext, useReducer, useEffect, useState, useContext } from "react";
+import {cartFunc} from './cartReducer'
+const axios = require('axios');
+
+export const CartContext = createContext();
+
+export const CartProvider = ({ children }) => {
+  const [cart, setCart] = useState([])
+  const [state, dispatch] = useReducer(cartFunc, cart);
+
+  useEffect(() => {
```

Comment:
add dependency array. Your get post request is not handled correctly please do all those changes.